pagopa/io-app
# fix: [IA-820] During a payment if the credit card insertions fails the payment remains `on-going`

## Short description
This PR runs the delete `payment activation procedure` when a credit card insertion fails

### before
https://user-images.githubusercontent.com/822471/162778508-2af5ff98-02f7-477f-b44a-7969f7efe1e9.mov

### after
https://user-images.githubusercontent.com/822471/162778586-791a792f-eacd-43f0-a024-f730b92b674a.mov

### how to test
- change [this outcome value](https://github.com/pagopa/io-dev-api-server/blob/master/src/config.ts#L98) with a number value `!== 0`
- try to pay 
- **just before the last step**, choose to add a new payment method, select credit card
- add a credit card, it will fail
- the `DELETE` API for the payment should be invoked

**NOTE**
- if the credit card insertion doesn't fail the `DELETE` API must not be invoked
